mobile/src/screens/Chess.tsx

```diff
@@ -1,28 +1,98 @@
 // Initial chessboard code credits go to William Candillon
 // Github: https://github.com/wcandillon
 // Source Code: https://github.com/wcandillon/can-it-be-done-in-react-native/tree/master/season4/src/Chess
-import { useRoute } from "@react-navigation/native";
-import React from "react";
-import { View, StyleSheet } from "react-native";
+import { useNavigation, useRoute } from "@react-navigation/native";
+import React, { useContext, useEffect, useState } from "react";
+import { View } from "react-native";
 import { gestureHandlerRootHOC } from "react-native-gesture-handler";
 
+import Button from "../components/Button";
 import Board from "../components/chess/Board";
-
-const styles = StyleSheet.create({
-  container: {
-    flex: 1,
-    justifyContent: "center",
-    backgroundColor: "grey",
-  },
-});
+import OneButtonPopup from "../components/popups/OneButtonPopup";
+import TwoButtonPopup from "../components/popups/TwoButtonPopup";
+import { SocketContext } from "../contexts/SocketContext";
+import { AppStylesheet as styles } from "../styles/AppStylesheet";
 
 // Enable gestures to work for Android
 const Chessboard = gestureHandlerRootHOC(() => {
+  const socket = useContext(SocketContext);
+  const navigation = useNavigation();
   const route = useRoute();
+  const [grayButton, setGrayButton] = useState(false);
+  // states for popups rendering
+  const [openDraw, setOpenDraw] = useState(false);
+  const [openDrawRejected, setOpenDrawRejected] = useState(false);
+  const [openDrawAccepted, setOpenDrawAccepted] = useState(false);
+
+  const proposeDraw = () => {
+    socket.emit("try draw");
+    setGrayButton(true);
+  };
+
+  const acceptDraw = () => {
+    socket.emit("draw accepted");
+    setOpenDraw(false);
+    navigation.navigate("HomeScreen");
+  };
+
+  const rejectDraw = () => {
+    socket.emit("draw rejected");
+    setOpenDraw(false);
+  };
+
+  useEffect(() => {
+    socket.on("draw request", () => {
+      setOpenDraw(true);
+    });
+
+    socket.on("game drawn", () => {
+      setOpenDrawAccepted(true);
+      setGrayButton(false);
+    });
+
+    socket.on("draw request rejected", () => {
+      setOpenDrawRejected(true);
+      setGrayButton(false);
+    });
+
+    // reset states when component unmounts
+    return () => {
+      setGrayButton(false);
+      setOpenDraw(false);
+      setOpenDrawRejected(false);
+      setOpenDrawAccepted(false);
+    };
+  }, []);
 
   return (
     <View style={styles.container}>
       <Board color={route.params.color} />
+      <Button
+        text="Draw"
+        onPress={grayButton ? undefined : proposeDraw}
```

Comment:
Just to clarify, the current functionality you've made with this is to disable the button until there is a response back on the already sent draw request?

Reply:
yup